core/src/main/java/com/linecorp/armeria/server/logging/LoggingService.java

```diff
@@ -37,72 +40,132 @@
+    static final String RESPONSE_FORMAT = "Response: {}";
 
     /**
      * Returns a new {@link Service} decorator that logs {@link Request}s and {@link Response}s at
      * {@link LogLevel#INFO}.
+     *
+     * @see LoggingServiceBuilder for more information on the default settings.
      */
     public static <I extends Request, O extends Response>
     Function<Service<I, O>, LoggingService<I, O>> newDecorator() {
-        return LoggingService::new;
+        return new LoggingServiceBuilder()
+                .requestLogLevel(LogLevel.INFO)
+                .successfulResponseLogLevel(LogLevel.INFO)
+                .failureResponseLogLevel(LogLevel.INFO)
+                .buildDecorator();
     }
 
     /**
-     * Returns a new {@link Service} decorator that logs {@link Request}s and {@link Response}s.
-     *
-     * @param level the log level
+     * @deprecated Use {@link LoggingServiceBuilder}.
      */
+    @Deprecated
     public static <I extends Request, O extends Response>
     Function<Service<I, O>, LoggingService<I, O>> newDecorator(LogLevel level) {
         return delegate -> new LoggingService<>(delegate, level);
     }
 
-    private final LogLevel level;
+    private final LogLevel requestLogLevel;
+    private final LogLevel successfulResponseLogLevel;
+    private final LogLevel failureResponseLogLevel;
+    private final Function<HttpHeaders, HttpHeaders> requestHeadersSanitizer;
+    private final Function<Object, Object> requestContentSanitizer;
+    private final Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer;
+    private final Function<Object, Object> responseContentSanitizer;
+    private final Sampler sampler;
 
     /**
-     * Creates a new instance that logs {@link Request}s and {@link Response}s at {@link LogLevel#INFO}.
+     * @deprecated Use {@link LoggingService#newDecorator()}.
      */
+    @Deprecated
     public LoggingService(Service<I, O> delegate) {
         this(delegate, LogLevel.INFO);
     }
 
     /**
-     * Creates a new instance that logs {@link Request}s and {@link Response}s at the specified
-     * {@link LogLevel}.
+     * @deprecated Use {@link LoggingServiceBuilder}.
      */
+    @Deprecated
     public LoggingService(Service<I, O> delegate, LogLevel level) {
-        super(delegate);
-        this.level = requireNonNull(level, "level");
+        this(delegate,
+             level,
+             level,
+             level,
+             Function.identity(),
+             Function.identity(),
+             Function.identity(),
+             Function.identity(),
+             Sampler.ALWAYS_SAMPLE);
+    }
+
+    /**
+     * Creates a new instance that logs {@link Request}s and {@link Response}s at the specified
+     * {@link LogLevel}s with the specified sanitizers.
+     */
+    LoggingService(
+            Service<I, O> delegate,
+            LogLevel requestLogLevel,
+            LogLevel successfulResponseLogLevel,
+            LogLevel failureResponseLogLevel,
```

Comment:
`successful` is not a noun while `failure` is. How about using adjective for both?